tests/test_port.py

```diff
@@ -61,20 +62,51 @@ def test_kneighbors(result, n_components):
 )
 @pytest.mark.parametrize("n_components", [None, 3], ids=["full", "reduced"])
 @pytest.mark.parametrize("weighted", [False, True], ids=["unweighted", "weighted"])
-def test_predict(result, n_components, weighted):
+@pytest.mark.parametrize("training", [True, False], ids=["training", "testing"])
```

Comment:
I agree that "training/testing" isn't quite right - I was queasy about this as well.  But I don't think "independent/dependent" captures this test.  The first is testing the reference data using independent neighbors, but the second is testing the target data, but is still using independent neighbors (because the target data wasn't used in `fit`).  How do you feel about reference/target instead?

Reply:
Ah, good point!

Reference/target sounds good to me - I like the consistency with the dataset attributes and file names.